src/vulkan/vk-utils.cpp

```
@@ -328,7 +328,30 @@ VkAccessFlagBits calcAccessFlags(ResourceState state)
+
+    if (api.m_deviceFeatures.tessellationShader)
+    {
+        result |=
+            VK_PIPELINE_STAGE_TESSELLATION_CONTROL_SHADER_BIT | VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT;
+    }
+
+    if (api.m_deviceFeatures.geometryShader)
+        result |= VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT;
+
+    if (api.m_extendedFeatures.rayTracingPipelineFeatures.rayTracingPipeline)
+        result |= VK_PIPELINE_STAGE_RAY_TRACING_SHADER_BIT_KHR;
```

Cppcheck (2.20.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)